pkg/api/protobuf/go/minder/v1/entities.go

```diff
@@ -29,6 +29,14 @@ const (
 	ArtifactEntity EntityType = "artifact"
 	// PullRequestEntity is a pull request entity
 	PullRequestEntity EntityType = "pull_request"
+	// ReleaseEntity is an entity abstracting a release
+	ReleaseEntity EntityType = "release"
+	// PipelineRunEntity is an entity abstracting a pipeline run (eg a workflow)
+	PipelineRunEntity EntityType = "pipeline_run"
+	// TaskRunEntity is an entity abstracting a task run (eg a step)
+	TaskRunEntity EntityType = "task_run"
+	// BuildEntity is an entity that represents a software build
+	BuildEntity EntityType = "build"
```

Comment:
@eleftherias - I think this is where @puerco put more info about which corresponds to what in GitHub terms. In that sense, I guess `task_run` might be a job instead of a step. And also if `pipeline_run` is the workflow, then what is the `build`? But I guess we don't have to define these now, right? We can do that once we start figuring out the specific provider implementation

Reply:
@eleftherias @rdimitrov yes, with these new events we should be able to write rules against the results of a Workflow (a PipelineRun in cd events terms), a Step (a TaskRun in cd events) and also to published releases.

The build itself in GitHub terms will look like a subclass of a Step run. It will share a lot of attributes but also has some particulars that differ from a regular Step run that we will need to check (dependencies, the builder, etc).

Reply:
Do you know what kind of rules we want to write against the results of a `PipelineRun` and `TaskRun`? I'm not yet seeing how they tie into the new rules we wanted to write and I want to make sure we're not adding entities that will  sit around unused.

Reply:
Perhaps it's difficult to add a specific answer, but my understanding is this should allows us to have more control over the software release lifecycle, i.e. at the end of this (workflow, job, step) we should have that, if not - block executing what's coming next.

Reply:
I guess another example would be to gather some metadata after each step of the delivery process and store it into an attestation.